packages/substrate-bindings/src/codecs/metadata/deprecation.ts

```diff
@@ -0,0 +1,26 @@
+import { _void, Enum, Option, str, Struct, Tuple, u8, Vector } from "scale-ts"
+
+export const itemDeprecation = Enum({
+  NotDeprecated: _void,
+  DeprecatedWithoutNote: _void,
+  Deprecated: Struct({
+    note: str,
+    since: Option(str),
+  }),
+})
+
+export const variantDeprecation = Vector(
+  Tuple(
+    u8,
+    Enum(
+      {
```

Comment:
I don't have context, but would it make sense to have this as a struct instead? To give a name to the `u8`